core/src/main/scala/zio/jdbc/ZConnection.scala

```diff
@@ -105,18 +116,20 @@ final class ZConnection(private[jdbc] val underlying: ZConnection.Restorable) ex
    * @param zc the connection to look into
    * @return true if the connection is alive (valid), false otherwise
    */
-  def isValid(timeout: Int): Task[Boolean] =
-    ZIO.attempt(this.underlying.isValid(timeout))
+  def isValid(timeout: Int): ZIO[Scope, FailedAccess[Boolean], Boolean] =
+    access(_.isValid(timeout))
 
   private[jdbc] def restore: UIO[Unit] =
     ZIO.succeed(underlying.restore())
 }
 
 object ZConnection {
 
-  def make(underlying: Connection): Task[ZConnection] =
+  def make(underlying: Connection): IO[ConnectionException, ZConnection] =
     for {
-      restorable <- ZIO.attempt(new Restorable(underlying))
+      restorable <- ZIO.attempt(new Restorable(underlying)).refineOrDie { case e: SQLException =>
+                      FailedMakingRestorable(e)
```

Comment:
Question: Do we expect `new Restorable(...)` to fail (it seems like a wrapper)?

Reply:
I think so, because when making a new instance of `Restorable` some methods that might throw `SQLException` of the underlying connection are called.